src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -199,9 +197,10 @@ private boolean authorizeRequest(RestHandler original, RestRequest request, Rest
         return true;
     }
 
-    private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel channel) throws Exception {
+    public boolean checkAndAuthenticateRequest(RestRequest request, RestChannel channel, ThreadContext restoringThreadContext)
+        throws Exception {
 
-        threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN, Origin.REST.toString());
+        restoringThreadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN, Origin.REST.toString());
```

Comment:
👏 This should make it easy to differentiate between regular thread context and restoring thread context and where they are used.

Reply:
Yea I noticed there was a mix of using the threadContext passed in as an arg vs the class-level threadContext. They are the same threadContext, but its confusing and the method should only use the threadContext passed in as an arg